### ISSUE TITLE
sidecar.istio.io/proxyCPU annotation erases resource limits

### ISSUE THREAD

**Issue by irl-segfault** [istio/istio#21240]
**Bug description**
using the `sidecar.istio.io/proxyCPU` pod annotation causes limits on the sidecar to be erased.  I appears that if that single annotation is used, the cpu limit, memory request, and memory limit are not taken from the default injection template. 

[ x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ x] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
the cpu request is set to the value of the annotation, but the cpu limit and memory request and memory limit are not affected, ideally still taken from the sidecar injection template.
**Steps to reproduce the bug**
deploy pod with the following annotation into and istio-enabled namespace
sidecar.istio.io/proxyCPU: "60m"
observe that no cpu limit, no memory request, and no memory limit is set on the sidecar
**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```$ istioctl version --remote
client version: 1.3.3
control plane version: 1.4.3

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.8-gke.3", GitCommit:"2ccf2c797713077f4a3d0f87c4b4bae725362326", GitTreeState:"clean", BuildDate:"2020-01-29T19:27:22Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by howardjohn** [comment]
Fixed by #22395. Thanks!

**Comment by StefanCenusa** [comment]
Hello, @howardjohn! 

I'm having this exact issue on **istio 1.8.1**. If I am setting the `sidecar.istio.io/proxyCPU` pod annotation, it causes the memory request/limits and cpu limits to be erased. I think the expected behaviour should be to inherit the default ones.

Default
```
    Limits:
      cpu:     2
      memory:  1Gi
    Requests:
      cpu:      100m
      memory:   128Mi
```
With `sidecar.istio.io/proxyCPU: 250m`
```
    Requests:
      cpu:      250m
```
Expected with `sidecar.istio.io/proxyCPU: 250m`
```
    Limits:
      cpu:     2
      memory:  1Gi
    Requests:
      cpu:      250m
      memory:   128Mi
```

Am I missing something? I see this issue is closed, but it doesn't seem to be resolved.

Thank you!
